code-studio/src/js/progress.js

```diff
@@ -192,13 +200,69 @@ progress.renderStageProgress = function (stageData, progressData, clientProgress
 };
 
 progress.renderCourseProgress = function (scriptData) {
-  var teacherCourse = $('#landingpage').hasClass('teacher-course');
+  var store = loadProgress(scriptData);
   var mountPoint = document.createElement('div');
+
+  $.ajax('/api/user_progress/' + scriptData.name).done(data => {
```

Comment:
We need to hit the API regardless.  For publicly-cached scripts we don't know if the user is logged in or not.  The right thing to do would be to implement an actual client-side user store so we have some idea of whether the user is logged in or not.